modules/nextflow/src/main/groovy/nextflow/trace/WebLogObserver.groovy

```diff
@@ -175,8 +181,9 @@ class WebLogObserver implements TraceObserver{
      * @param event The current run status. One of {'started', 'process_submit', 'process_start',
      * 'process_complete', 'error', 'completed'}
      * @param trace A TraceRecord object that contains current process information
+     * @param manifest A Map with the manifest information
      */
-    protected void sendHttpMessage(String event, TraceRecord trace = null){
+    protected void sendHttpMessage(String event, TraceRecord trace = null, Map manifest = null){
 
```

Comment:
Since `trace` and `manifest` are mutually exclusive, what about passing instead a singe `payload` generic argument, then using something like 

```groovy
if( payload instanceof TraceRecord ) { .. } 
else if( payload instanceof Manifest ) { .. } 
else throw new IllegalArgumentException(..)
```

Reply:
Uh that is nice! I was not yet fully happy with my solution

Reply:
``` 
protected void sendHttpMessage(String event, Object payload = null){

        // Set the message info
        final time = new Date().format("yyyy-MM-dd'T'HH:mm:ss'Z'", UTC)

        final message = new HashMap(5)
        message.runName = runName
        message.runId = runId
        message.event = event
        message.runStatus = event // deprecated to be removed
        message.utcTime = time

        if (payload) {
            if (payload instanceof TraceRecord)
                message.trace = payload.store
            else if (payload instanceof Manifest)
                message.manifest = manifest
            else throw new IllegalArgumentException("Only TraceRecord and Manifest class types are supported.")
        }
...
```

Reply:
It seems good, hopefully tested :D

Reply:
But I am in the same package, and `store` has package scope. So I am missing something here...

Reply:
it's also useful to report the object type in the exception message: 

```groovy 
            if (payload instanceof TraceRecord)
                message.trace = payload.store
            else if (payload instanceof Manifest)
                message.manifest = manifest
            else if( payload != null ) 
                 throw new IllegalArgumentException("Only TraceRecord and Manifest class types are supported: [${payload.getClass().getName()}] $payload")

```

Reply:
> But I am in the same package, and `store` has package scope. So I am missing something here...

Not sure to understand ?

Reply:
It does not compile :( throws `Access to nextflow.trace.TraceRecord#store is forbidden @ line 194, column 47.`

Reply:
ummm, double check the test is in the same package

Reply:
I am running `compile.sh`, to check with a local REST server, if it works correctly. There it fails.-

Reply:
stack trace ?

Reply:
https://pastebin.com/0RRa0MNh

Sorry, thats all I get.

Reply:
So removing the `@CompileStatic` annotation of the `WeblogObserver` fixed it for now. So can it be, that the Class type of payload is `Object` at compile time (which causes the exception during compilation), but is evaluated as either `TraceRecord` or `Manifest` at runtime.  Does this make sense? Not sure...

Reply:
A bit bit confused: I've pulled this branch and I could compile w/o any problem. Then the .store property is accessed also in the current implementation (I mean the one in the master). Why it should fail now?

(sorry reposted in the correct thread)